File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/landing/SubscriptionRow.scala
package dev.typr.foundationssc.docs.landing

import dev.typr.foundationssc.data.*
import java.time.Instant
import java.util.UUID

@SuppressWarnings(Array("unused"))
object SubscriptionRow:
  enum Plan:
    case free, pro, team

  // start
  case class Subscription(
      id: UUID,
      email: String,
      plan: Plan,
      activeRange: Range[Instant],
      metadata: Option[Jsonb],
      cancelledAt: Option[Instant]
  )
  // stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/landing/SubscriptionRow.java
package dev.typr.foundations.docs.landing;

import dev.typr.foundations.data.Jsonb;
import dev.typr.foundations.data.Range;
import java.time.Instant;
import java.util.Optional;
import java.util.UUID;

@SuppressWarnings("unused")
public class SubscriptionRow {
  enum Plan { free, pro, team }

  // start
  record Subscription(
      UUID id,
      String email,
      Plan plan,
      Range<Instant> activeRange,
      Optional<Jsonb> metadata,
      Optional<Instant> cancelledAt
  ) {}
  // stop
}
